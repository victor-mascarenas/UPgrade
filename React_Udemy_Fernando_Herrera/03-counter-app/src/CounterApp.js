import React, {useState} from "react";
import { PropTypes } from "prop-types";

const CounterApp = ({value}) => {
    const [counter, setCounter] = useState(value);
    const plusBtnOnClick = () => {
        setCounter(counter + 1);
    };
    const resetBtnOnClick = () => {
        setCounter(value);
    };
    const minusBtnOnClick = () => {
        setCounter(counter - 1);
    };
    return (
        <div>
            <h1>Counter App</h1>
            <h2>{counter}</h2>
            <button id="plusBtn" onClick={plusBtnOnClick}>+1</button>
            <button id="resetBtn" onClick={resetBtnOnClick}>Reset</button>
            <button id="minusBtn" onClick={minusBtnOnClick}>-1</button>
        </div>
    );
}
CounterApp.propTypes = {
    value: PropTypes.number
}
CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;