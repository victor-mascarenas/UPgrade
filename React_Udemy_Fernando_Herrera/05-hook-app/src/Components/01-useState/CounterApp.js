import React, { useState } from 'react';
import './counter.css';

const CounterApp = () => {
    const [state, setState] = useState({
        counter1: 10,
        counter2: 20
    });
    const {counter1, counter2} = state;

    const buttonOnClik = () => {
        setState({
            ...state,
            counter1: counter1 + 1
        });
    }

    return (
        <>
            <h1>Counter1 {counter1}</h1>
            <h1>Counter2 {counter2}</h1>
            <hr/>
            <button className='btn btn-primary' onClick={buttonOnClik}>+ 1</button>
        </>
    );
};

export default CounterApp;