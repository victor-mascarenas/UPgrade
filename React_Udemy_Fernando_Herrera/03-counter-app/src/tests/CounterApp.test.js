import React from "@testing-library/react";
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp/>', () => {
    test('Debe ser igual a la snapshot (sin enviar valor)', () => {
        const wrapper = shallow(<CounterApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe mostrar el valor por defecto de 0', () => {
        const valor = "0";
        const wrapper = shallow(<CounterApp/>);
        const valorMostrado = wrapper.find('h2').text();
        expect(valorMostrado).toBe(valor);
    });
    test('Debe mostrar el valor de 100', () => {
        const valor = "100";
        const wrapper = shallow(<CounterApp value={100}/>);
        const valorMostrado = wrapper.find('h2').text();
        expect(valorMostrado).toBe(valor);
    });
});
