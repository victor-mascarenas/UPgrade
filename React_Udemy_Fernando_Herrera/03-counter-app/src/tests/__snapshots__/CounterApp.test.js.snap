// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> Debe ser igual a la snapshot (sin enviar valor) 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    id="plusBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="resetBtn"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="minusBtn"
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
